src/coreComponents/mesh/generators/VTKUtilities.hpp

```diff
@@ -0,0 +1,576 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file VTKUtilities.hpp
+ */
+
+#ifndef GEOSX_MESH_GENERATORS_VTKUTILITIES_HPP
+#define GEOSX_MESH_GENERATORS_VTKUTILITIES_HPP
+
+#include "common/DataTypes.hpp"
+#include "common/DataLayouts.hpp"
+#include "common/MpiWrapper.hpp"
+#include "common/TypeDispatch.hpp"
+#include "mesh/DomainPartition.hpp"
+#include "mesh/MeshBody.hpp"
+#include "mesh/generators/CellBlockManager.hpp"
+#include "mesh/generators/VTKMeshGeneratorTools.hpp"
+#include "mesh/generators/ParMETISInterface.hpp"
+#ifdef GEOSX_USE_SCOTCH
+#include "mesh/generators/PTScotchInterface.hpp"
+#endif
+
+#include <vtkSmartPointer.h>
+#include <vtkDataSet.h>
+#include <vtkCellArray.h>
+#include <vtkMultiProcessController.h>
+#include <vtkBoundingBox.h>
+#include <vtkDataArray.h>
+#include <vtkPointData.h>
+#include <vtkPartitionedDataSet.h>
+#include <vtkExtractCells.h>
+
+#include <numeric>
+#include <unordered_set>
+
+namespace geosx
+{
+
+using namespace dataRepository;
+
+/**
+ * @brief Choice of advanced mesh partitioner
+ */
+enum class PartitionMethod : integer
+{
+  parmetis, ///< Use ParMETIS library
+  ptscotch, ///< Use PTScotch library
+};
+
+/// Strings for VTKMeshGenerator::PartitionMethod enumeration
+ENUM_STRINGS( PartitionMethod,
+              "parmetis",
+              "ptscotch" );
+
+/**
+ * @brief Type of map used to store cell lists.
+ *
+ * This should be an unordered_map, but some outdated standard libraries on some systems
+ * do not provide std::hash specialization for enums. This is not performance critical though.
+ */
+using CellMapType = std::map< ElementType, std::unordered_map< int, std::vector< vtkIdType > > >;
```

Comment:
Perhaps all of these could also be in the `geosx::vtk::` namespace?

Reply:
indeed